src/player.js

```diff
@@ -1876,6 +1943,20 @@ export default class Player extends FakeEventTarget {
     }
   }
 
+  /**
+   *
+   * add a text track to the track array
+   * @param {TextTrack} track - the track to be added to the track array
+   * @returns {void}
+   * @public
+   */
+  addTextTrack(track: TextTrack): void {
+    // allow addition of textTracks only.
+    if (track instanceof TextTrack) {
+      this._tracks.push(track);
```

Comment:
Don't push it before engine is actually adding this track.
You should call engine.addTextTrack and just if the action succeeded do it.

Reply:
removed this API